diff --git a/src/ucd-aws.c b/src/ucd-aws.c
index ded0bf0..ceedf1f 100644
--- a/src/ucd-aws.c
+++ b/src/ucd-aws.c
@@ -58,7 +58,13 @@
 #define AWS_IP "169.254.169.254"
 #define AWS_REQUEST_SSHKEY "GET /latest/meta-data/public-keys/0/openssh-key HTTP/1.0\r\nhost: " AWS_IP "\r\nConnection: keep-alive\r\n\r\n"
 #define AWS_REQUEST_USERDATA "GET /latest/user-data HTTP/1.0\r\nhost: " AWS_IP "\r\nConnection: close\r\n\r\n"
-#define CLOUD_CONFIG_SSH_HEADER "#cloud-config\nssh_authorized_keys:\n  - "
+#define CLOUD_CONFIG_SSH_HEADER \
+"#cloud-config\n" \
+"users:\n" \
+"  - name: clear\n"\
+"    groups: wheel\n"\
+"    ssh-authorized-keys:\n"\
+"      - "
 
 
 /*
@@ -87,6 +93,8 @@ static int parse_headers(FILE *f, long int *cl)
 			if (errno == EINVAL || errno == ERANGE) {
 				return 0;
 			}
+	        } else if (strstr(buf, "404 Not Found")) {
+			return 2;
 		} else if (strncmp(buf, "HTTP/1.0", 8) == 0) {
 			long int status = strtol(&buf[8], NULL, 10);
 			if (errno == EINVAL || errno == ERANGE) {
