--- micro-config-drive-27/src/ucd-aws.c~	2017-04-28 21:46:28.000000000 +0000
+++ micro-config-drive-27/src/ucd-aws.c	2017-05-05 00:47:58.140019624 +0000
@@ -78,8 +78,8 @@
 		int r = connect(sockfd, (struct sockaddr *)&server, sizeof(server));
 		if (r == 0)
 			break;
-		if ((r != EAGAIN) && (r != ENETUNREACH) && (r != ETIMEDOUT))
-			FAIL("connect()");
+//		if ((r != EAGAIN) && (r != ENETUNREACH) && (r != ETIMEDOUT))
+//			FAIL("connect()");
 		nanosleep(&ts, NULL);
 		if (++n > 200) /* 10 secs */
 			FAIL("timeout in connect()");
